mcstatus/forge_data.py

```diff
@@ -0,0 +1,257 @@
+"""Decoder for data from Forge, that is included into a response object.
+
+After 1.18.1, Forge started to compress its mod data into a
+UTF-16 string that represents binary data containing data like
+the forge mod loader network version, a big list of channels
+that all the forge mods use, and a list of mods the server has.
+
+Before 1.18.1, the mod data was in `forgeData` attribute inside
+a response object. We support this implementation too.
+
+For more information see this file from forge itself:
+https://github.com/MinecraftForge/MinecraftForge/blob/54b08d2711a15418130694342a3fe9a5dfe005d2/src/main/java/net/minecraftforge/network/ServerStatusPing.java#L27-L73
+"""
+
+from __future__ import annotations
+
+import io
+from dataclasses import dataclass
+from typing import Final, IO, TYPE_CHECKING
+
+from mcstatus.protocol.connection import BaseConnection, BaseReadSync, Connection
+
+VERSION_FLAG_IGNORE_SERVER_ONLY: Final = 0b1
+IGNORE_SERVER_ONLY: Final = "<not required for client>"
+
+
+if TYPE_CHECKING:
+    from typing_extensions import NotRequired, Self, TypedDict
+
+    class RawForgeDataChannel(TypedDict):
+        res: str
+        """Channel name and ID (for example ``fml:handshake``)."""
+        version: str
+        """Channel version (for example ``1.2.3.4``)."""
+        required: bool
+        """Is this channel required for client to join?"""
+
+    class RawForgeDataMod(TypedDict):
+        modid: NotRequired[str]
+        modId: NotRequired[str]  # camel case
```

Comment:
those comments were here to explain `noqa` comment, but [ruff ignores N815 (mixed-case-variable-in-class-scope) for typed dicts](https://github.com/astral-sh/ruff/issues/4014), so there is no error. this comment doesn't have sense anymore

Reply:
Removed in 7675af1